### PR TITLE
HADOOP-16237. Fix new findbugs issues after update guava to 27.0-jre …

### PR DESCRIPTION
**Pull request by bgaborg**
…in hadoop-project trunk

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 25 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 76 | Maven dependency ordering for branch |
| +1 | mvninstall | 1024 | trunk passed |
| +1 | compile | 974 | trunk passed |
| +1 | checkstyle | 187 | trunk passed |
| +1 | mvnsite | 92 | trunk passed |
| +1 | shadedclient | 935 | branch has no errors when building and testing our client artifacts. |
| -1 | findbugs | 42 | hadoop-common-project/hadoop-kms in trunk has 1 extant Findbugs warnings. |
| +1 | javadoc | 49 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 23 | Maven dependency ordering for patch |
| +1 | mvninstall | 56 | the patch passed |
| +1 | compile | 913 | the patch passed |
| +1 | javac | 913 | the patch passed |
| +1 | checkstyle | 187 | the patch passed |
| +1 | mvnsite | 67 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 4 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 620 | patch has no errors when building and testing our client artifacts. |
| +1 | findbugs | 44 | hadoop-common-project/hadoop-kms generated 0 new + 0 unchanged - 1 fixed = 0 total (was 1) |
| +1 | findbugs | 58 | ozone-recon in the patch passed. |
| +1 | javadoc | 47 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 207 | hadoop-kms in the patch passed. |
| +1 | unit | 46 | ozone-recon in the patch passed. |
| +1 | asflicense | 39 | The patch does not generate ASF License warnings. |
| | | 5721 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-710/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/710 |
| Optional Tests |  dupname  asflicense  xml  compile  javac  javadoc  mvninstall  mvnsite  unit  shadedclient  findbugs  checkstyle  |
| uname | Linux 1d070e620bd5 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / bb8dda2 |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| findbugs | v3.1.0-RC1 |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-710/1/artifact/out/branch-findbugs-hadoop-common-project_hadoop-kms-warnings.html |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-710/1/testReport/ |
| Max. process+thread count | 413 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-kms hadoop-ozone/ozone-recon U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-710/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by bgaborg** [comment]
first run looks great, fixes the an issue. let's see if the new one fixes two.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 31 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 64 | Maven dependency ordering for branch |
| +1 | mvninstall | 1042 | trunk passed |
| +1 | compile | 990 | trunk passed |
| +1 | mvnsite | 305 | trunk passed |
| +1 | shadedclient | 3042 | branch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-yarn-project/hadoop-yarn |
| -1 | findbugs | 36 | hadoop-common-project/hadoop-kms in trunk has 1 extant Findbugs warnings. |
| +1 | javadoc | 143 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 24 | Maven dependency ordering for patch |
| +1 | mvninstall | 277 | the patch passed |
| +1 | compile | 930 | the patch passed |
| +1 | javac | 930 | the patch passed |
| +1 | mvnsite | 309 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 4 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 599 | patch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-yarn-project/hadoop-yarn |
| +1 | findbugs | 37 | hadoop-common-project/hadoop-kms generated 0 new + 0 unchanged - 1 fixed = 0 total (was 1) |
| +1 | javadoc | 137 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 227 | hadoop-kms in the patch passed. |
| -1 | unit | 7883 | hadoop-yarn in the patch failed. |
| +1 | asflicense | 43 | The patch does not generate ASF License warnings. |
| | | 13862 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.yarn.server.resourcemanager.TestKillApplicationWithRMHA |
|   | hadoop.yarn.server.resourcemanager.metrics.TestCombinedSystemMetricsPublisher |
|   | hadoop.yarn.server.resourcemanager.TestRMEmbeddedElector |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-710/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/710 |
| Optional Tests |  dupname  asflicense  xml  compile  javac  javadoc  mvninstall  mvnsite  unit  shadedclient  findbugs  |
| uname | Linux 2b63d7b432a7 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 2d4f6b6 |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| findbugs | v3.1.0-RC1 |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-710/3/artifact/out/branch-findbugs-hadoop-common-project_hadoop-kms-warnings.html |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-710/3/artifact/out/patch-unit-hadoop-yarn-project_hadoop-yarn.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-710/3/testReport/ |
| Max. process+thread count | 927 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-kms hadoop-yarn-project/hadoop-yarn U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-710/3/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by bgaborg** [comment]
We don't need the commas at the end of the comments here. I added those just to trigger findbugs on yetus: 

> Skipped patched modules with no Java source: hadoop-yarn-project/hadoop-yarn

**Comment by bgaborg** [comment]
last patch should handle all the issues left

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 29 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 22 | Maven dependency ordering for branch |
| +1 | mvninstall | 985 | trunk passed |
| +1 | compile | 987 | trunk passed |
| +1 | mvnsite | 296 | trunk passed |
| +1 | shadedclient | 2905 | branch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-yarn-project/hadoop-yarn |
| -1 | findbugs | 30 | hadoop-common-project/hadoop-kms in trunk has 1 extant Findbugs warnings. |
| +1 | javadoc | 138 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 22 | Maven dependency ordering for patch |
| +1 | mvninstall | 263 | the patch passed |
| +1 | compile | 942 | the patch passed |
| +1 | javac | 942 | the patch passed |
| +1 | mvnsite | 319 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 5 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 648 | patch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-yarn-project/hadoop-yarn |
| +1 | findbugs | 38 | hadoop-common-project/hadoop-kms generated 0 new + 0 unchanged - 1 fixed = 0 total (was 1) |
| +1 | javadoc | 140 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 197 | hadoop-kms in the patch passed. |
| -1 | unit | 7505 | hadoop-yarn in the patch failed. |
| +1 | asflicense | 42 | The patch does not generate ASF License warnings. |
| | | 13337 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-710/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/710 |
| Optional Tests |  dupname  asflicense  xml  compile  javac  javadoc  mvninstall  mvnsite  unit  shadedclient  findbugs  |
| uname | Linux a7726e10d301 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 73f43ac |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| findbugs | v3.1.0-RC1 |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-710/5/artifact/out/branch-findbugs-hadoop-common-project_hadoop-kms-warnings.html |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-710/5/artifact/out/patch-unit-hadoop-yarn-project_hadoop-yarn.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-710/5/testReport/ |
| Max. process+thread count | 896 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-kms hadoop-yarn-project/hadoop-yarn U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-710/5/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-timelineservice-documentstore/src/main/java/org/apache/hadoop/yarn/server/timelineservice/documentstore/collection/document/flowrun/FlowRunDocument.java [hunk 1]
```diff
@@ -136,6 +140,8 @@ private TimelineMetric aggregate(TimelineMetric incomingMetric,
           .aggregate(incomingMetric, baseMetric, null);
     default:
       //NoOP
```
Comment (jojochuang) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-timelineservice-documentstore/src/main/java/org/apache/hadoop/yarn/server/timelineservice/documentstore/collection/document/flowrun/FlowRunDocument.java]: Please remove it.

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-timelineservice-documentstore/src/main/java/org/apache/hadoop/yarn/server/timelineservice/documentstore/reader/cosmosdb/CosmosDBDocumentStoreReader.java [hunk 2]
```diff
@@ -49,7 +49,7 @@
       .getLogger(CosmosDBDocumentStoreReader.class);
   private static final int DEFAULT_DOCUMENTS_SIZE = 1;
 
-  private static DocumentClient client;
+  private static volatile DocumentClient client;
```
Comment (jojochuang) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-timelineservice-documentstore/src/main/java/org/apache/hadoop/yarn/server/timelineservice/documentstore/reader/cosmosdb/CosmosDBDocumentStoreReader.java]: The volatile keyword is used to address double-checked-locking issue. Neat.

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-timelineservice-documentstore/src/main/java/org/apache/hadoop/yarn/server/timelineservice/documentstore/collection/document/flowrun/FlowRunDocument.java (hunk 1 above)
Comment (bgaborg) [reply hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-timelineservice-documentstore/src/main/java/org/apache/hadoop/yarn/server/timelineservice/documentstore/collection/document/flowrun/FlowRunDocument.java]: ok

**Comment by bgaborg** [comment]
thanks for the review. I've updated the pr.

**Comment by steveloughran** [comment]
committed to trunk

**Comment by bgaborg** [comment]
committed to trunk from a patch

**Pull request closed without merging**